time_stamper_cv/src/led_detection_node.cpp

```diff
@@ -16,7 +17,13 @@ void LedDetectionNode::start() {
 }
 
 void LedDetectionNode::callbackRawImage(const sensor_msgs::Image &image) const {
-  cv_bridge::CvImage out_msg = calibration_->process(image);
+  ROS_INFO_ONCE("Received first image");
+  cv_bridge::CvImage out_msg = image_processor_->process(image);
+
+  time_stamper_cv::Ledstate  led_msg = image_processor_->getLedStateMessage();
+  led_msg.header = image.header;
```

Comment:
I find it problematic that `getLedStateMessage` kind of claims to produce a complete message but the caller still has to patch in the head. What about going the other way around: the caller of `getLedStateMessage` creates the message and hands passes it on in a mutable way to it (e.g non-const ref). That would of course require a renaming of the function, to e.g. `fillInLedState`. I really don't like "output" parameters but returning something incomplete I find even worse :D. In an ideal world the message would be comprised of bigger junks, one of which this message would return. But that isn't really easy to achieve here, I believe.

That suggested pattern of a `fillIn..` function would also solve the problem of the `is_valid`: it could be set by the caller (i.e. here or in the `process` function in case you move this into it).